controllers/spec.go

```diff
@@ -84,6 +84,14 @@ func (s *ContainerSpec) setArgs(arg ...string) *ContainerSpec {
 }
 
 func (s *ContainerSpec) addEnv(name, value string) *ContainerSpec {
+	// find existing EnvVar with same name
```

Comment:
This seems counterintuitive to general logic of adding an environment variable with an existing key. Since only the Java OPTS environment variable requires appending to an existing variable, can we make that logic specific to the system metrics service?

Reply:
Yes, removed changes from here and added to function responsible for adding system metrics sidecar.